core/templates/pages/blog-dashboard-page/blog-dashboard-tile/blog-dashboard-tile.component.ts

```diff
@@ -47,6 +47,8 @@ export class BlogDashboardTileComponent implements OnInit {
 
   ngOnInit(): void {
     const lastUpdated = this.blogPostSummary.lastUpdated;
+    // Last updated date is undefined if the blog post is published but not
+    // yet updated. In that case, we do not display the last updated date.
```

Comment:
This comment makes it sound like 'undefined' is valid, but it contradicts the fact that you're throwing an error here.

Can you substantiate (give proof) for your claim that the last-updated date is undefined in this scenario?

Reply:
[here ](https://github.com/oppia/oppia/blob/f05d81f3f48bcd8dc2b2302835f59bb0703166f3/core/templates/domain/blog/blog-post-summary.model.ts#L71) 

Yes using undefine is valid, but if last updated is not yet done, we will throw error.

Reply:
[here ](https://github.com/oppia/oppia/blob/f05d81f3f48bcd8dc2b2302835f59bb0703166f3/core/templates/domain/blog/blog-post-summary.model.ts#L71) 

Yes using undefine is valid, but if last updated is not yet done, we will throw error.

Reply:
The link you've pointed to isn't sufficient justification. The type annotation there might be wrong.

Either undefined is valid (in which case there shouldn't be an error thrown) or it is not (in which case undefined should not be allowed). To figure this out you would need to analyze the flow for how this property is set and used.

Reply:
Okay issue in [#13637](https://github.com/oppia/oppia/issues/13637) Blog post is related to it. we are creating an interstitial blog post that would be displayed in the editor until the actual blog post is fetched from the backend. In that interstitial blog post last updated is undefine. 

Blog post data is used to create blog post summary in which we are set last update as undefined.

Add todo to remove check after [#13637](https://github.com/oppia/oppia/issues/13637) is done

Reply:
I don't see the point of doing the wrong thing first and the right thing later. I suggest removing that file from this PR, and tackling it later when you're ready to tackle #13637.

Reply:
Done, Removed.